lib/cli/src/Cardano/CLI.hs

```diff
@@ -451,29 +454,37 @@ fullKeyEncodingDescription = mconcat
     , "- xpub1... (bech32-encoded extended public key)"
     ]
 
-xprvFromBytes :: ByteString -> Either String XPrvOrXPub
-xprvFromBytes = fmap AXPrv . left showErr . xPrvFromStrippedPubXPrvCheckRoundtrip
+aXPrvFromBytes :: ByteString -> Either String XPrvOrXPub
+aXPrvFromBytes x = maybe errLength (return . AXPrv) $ xprvFromBytes x
   where
-    showErr (ErrInputLengthMismatch expected actual) = mconcat
+    actual = BS.length x
+    expectedBytes = 96 :: Int
+    errLength = Left $ mconcat
         [ "Expected extended private key to be "
-        , show expected
+        , show expectedBytes
         , " bytes but got "
         , show actual
         , " bytes."
         ]
-    showErr (ErrCannotRoundtripToSameBytes) = mconcat
-        [ "That extended private key looks weird. "
-        , "Is it encrypted? Or is it an old Byron key?"
-        ]
 
-xpubFromBytes :: ByteString -> Either String XPrvOrXPub
-xpubFromBytes = fmap AXPub . CC.xpub
+aXPubFromBytes :: ByteString -> Either String XPrvOrXPub
+aXPubFromBytes x = maybe errLength (return . AXPub) $ xpubFromBytes x
+  where
+    actual = BS.length x
+    expectedBytes = 64 :: Int
```

Comment:
`64` as the length of public key with chain code probably also.
the same for `32` as public key without chain code.

Reply:
> 32 as public key without chain code

Not sure we ever deal with those?